pre_commit_hooks/trailing_whitespace_fixer.py

```diff
@@ -1,24 +1,28 @@
 from __future__ import print_function
 
 import argparse
-import fileinput
 import os
 import sys
+import tempfile
 
 from pre_commit_hooks.util import cmd_output
 
 
 def _fix_file(filename, markdown=False):
-    for line in fileinput.input([filename], inplace=True):
-        # preserve trailing two-space for non-blank lines in markdown files
-        if markdown and (not line.isspace()) and (line.endswith("  \n")):
-            line = line.rstrip(' \n')
-            # only preserve if there are no trailing tabs or unusual whitespace
-            if not line[-1].isspace():
-                print(line + "  ")
-                continue
-
-        print(line.rstrip())
+    with tempfile.NamedTemporaryFile(mode='w+b', delete=False) as tmp_file:
+        with open(filename, mode='rb') as original_file:
+            for line in original_file.readlines():
+                # preserve trailing two-space for non-blank lines in markdown files
+                eol = b'\r\n' if line[-2:] == b'\r\n' else b'\n'
+                if markdown and (not line.isspace()) and line.endswith(b'  ' + eol):
+                    line = line.rstrip(b' \r\n')  # restricted stripping: e.g. \t are not stripped
```

Comment:
either way, I wouldn't worry about this if it means changing a test.  We can address that in a followup or not at all :)